app/uk/gov/hmrc/gform/models/FormModel.scala

```diff
@@ -146,8 +146,9 @@ case class FormModel[A <: PageMode](
     first.fold(illegal) {
       case FormCtx(formComponentId) => explicitTypedExpr(expr, formComponentId)
       case DateCtx(_)               => TypeInfo(expr, StaticTypeData(ExprType.dateString, None))
-      case IsNumberConstant(_)      => TypeInfo(expr, StaticTypeData(ExprType.number, Some(Number())))
-      case otherwise                => TypeInfo(expr, StaticTypeData(ExprType.string, None))
+      case IsNumberConstant(d) =>
+        TypeInfo(expr, StaticTypeData(ExprType.number, Some(Number(maxFractionalDigits = d.scale))))
```

Comment:
This seems like a sane thing to do and actually is not wrong, but due to the lack of any deeper expression specification wrt literal numbers, changing this will have potential effect across all the forms. So my question would be what is the motivation for this change?

It doesn't seems to be connected with GFORMS-1097 at all.